functions/max_active_instances.pp

```diff
@@ -0,0 +1,25 @@
+# Provides a reasonable calculation for the maximum number of active instances
+# for a system
+#
+# Parameters are not to be used but are present as an assist to testing
+#
+# @param memory_limited_instances
+# @param processor_count
+#
+# @return [Integer]
+#   The maximum number of instances that should be active on the ``puppetserver``
+#
+function pupmod::max_active_instances (
+  Integer[0] $memory_limited_instances = Integer(inline_template("<%= @facts['memorysize_mb'].to_i / 512 %>")),
```

Comment:
You can do this sort of math in the puppet language now, no reason to use inline_template

Reply:
:+1: